sdk/storage/Azure.Storage.Files.Shares/perf/Azure.Storage.Files.Shares.Perf/Infrastructure/PerfTestEnvironment.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core.TestFramework;
+
+namespace Azure.Storage.Files.Shares.Perf
+{
+    /// <summary>
+    /// Represents the ambient environment in which the test suite is being run, offering access to information such as environment variables.
+    /// </summary>
+    internal sealed class PerfTestEnvironment : TestEnvironment
+    {
+        /// <summary>
+        /// The shared instance of the <see cref="PerfTestEnvironment"/> to be used during test runs.
+        /// </summary>
+        public static PerfTestEnvironment Instance { get; } = new PerfTestEnvironment();
+
+        /// <summary>
+        ///   The storage account endpoint suffix of the cloud to use for testing.
+        /// </summary>
+        public new string StorageEndpointSuffix => base.StorageEndpointSuffix ?? "core.windows.net";
```

Comment:
Are there cases where the base `StorageEndpointSuffix` is not set?

Reply:
I don't know, this is borrowed from an existing test. I can remove this and check the behavior.

Reply:
This is an attempt to reduce friction in common excecution scenarios.  Generally, it saves environment configuration for any run not bound to execution of `test-resources.json` - which right now is all of them.  This was something that you and I had discussed for inclusion in the Core test environment at one point but ultimately decided against adding.

Reply:
Why don't we use the `test-resources.json`?

Reply:
Because this isn't executing in the pipelines at present, it's all stand-alone in a VM using pre-existing Azure resources.   @mikeharder is still working on some kind of runner to integrate with pipelines and run on a schedule.

Reply:
Hm, but `test-resources.json` doesn't require the pipeline. In my understanding, we can run New-TestResources on the stand-alone VM same way we do for local machines.

I'm OK leaving it for now but think `test-resources.json` is crucial for someone who wants to run tests for themselves and try to reproduce the results.

Reply:
Nothing precludes running test resources; this just makes it easier to get started and iterate since our F5 experience is.... not ideal.  In this particular case, the storage test resources provision a ton of things that aren't relevant to the performance tests.

Reply:
Let's discuss it as part of infrastructure improvements for perf tests.